### PR TITLE
Add script to report status of RAPIDS repos

### PR DESCRIPTION
**Pull request by gmarkall**
Inspired by the [`repo status`](https://source.android.com/setup/develop/repo#status) command.

By default the short form of the status is reported, along with the branch name (`-s -b`) - an alternative form can be reported by providing the arguments to `git status` as arguments to this script.

Some use cases I have for this are:

- To quickly see if I'm a long way behind upstream after running `scripts/fetch-rapids-repositories.sh`,
- prior to switching branches to check I don't have any local modifications anywhere,
- when I come back to some work after a break to remind myself of what's in-flight.

Example output:

status-example

### CODE REVIEW

File: scripts/status-rapids-repositories.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+if [ ! -z $1 ]
+then
+    # Use arguments for status from user
+    ARGS="$@"
+else
+    # Default arguments: short-form changes with branch
+    ARGS="-s -b"
+fi
+
+cd $(dirname "$(realpath "$0")")/../../
```
Comment (bdice) [inline scripts/status-rapids-repositories.sh:20]: Does this need to be escaped? (I'm not 100% sure.)
```suggestion
cd $(dirname "$(realpath \"$0\")")/../../
```

File: scripts/status-rapids-repositories.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+BASE_DIR="$(pwd)"
+
+CODE_REPOS="${CODE_REPOS:-rmm raft cudf cuml cugraph cuspatial}"
+ALL_REPOS="${ALL_REPOS:-$CODE_REPOS compose notebooks-contrib}"
+
+for REPO in $ALL_REPOS; do
+    echo "$REPO:"
+    cd "$BASE_DIR/$REPO";
+    git status $ARGS;
+    cd - >/dev/null 2>&1;
```
Comment (bdice) [inline scripts/status-rapids-repositories.sh]: Is the last `cd -` command being noisy or is this redirecting the `git status` output? Does this help?
```suggestion
    pushd "$BASE_DIR/$REPO"
    git status $ARGS > /dev/null 2>&1
    popd
```

File: scripts/status-rapids-repositories.sh (hunk 1 above)
Comment (gmarkall) [reply scripts/status-rapids-repositories.sh:20]: I'm not sure, and I have to confess my bash expertise is quite limited. The heritage of this script is that it's derived from [`pull-rapids-repositories.sh`](https://github.com/trxcllnt/rapids-compose/blob/master/scripts/pull-rapids-repositories.sh) which I didn't author, and doesn't escape those quotes, for whatever that's worth :slightly_smiling_face:

File: scripts/status-rapids-repositories.sh (hunk 2 above)
Comment (gmarkall) [reply scripts/status-rapids-repositories.sh]: `cd -` is noisy - it prints the directory that was changed to. I'll look into `pushd` and `popd` though.

Comment (gmarkall) [reply scripts/status-rapids-repositories.sh]: About the git status - the output doesn't want to be redirected, because it's needed to appear on the terminal.

**Comment by gmarkall** [comment]
@bdice thanks for having a look at this - I've incorporated the suggestion to use `pushd` and `popd`. I'm not even sure how to figure out whether to escape the quotes though - can you share any arguments either way?

File: scripts/status-rapids-repositories.sh (hunk 1 above)
Comment (bdice) [reply scripts/status-rapids-repositories.sh:20]: It's probably fine to leave this as-is. If there are problems, we can fix it later.

**Review by bdice — APPROVED** [review]
Looks fine to me!

File: scripts/status-rapids-repositories.sh (hunk 1 above)
Comment (bdice) [reply scripts/status-rapids-repositories.sh:20]: If you wanted to test an edge case, it would probably involve paths with spaces.

**Review by trxcllnt — APPROVED** [review]
